### PR TITLE
Feature/signalfx support

### PR DESCRIPTION
**Pull request by nicmunroe**

### CODE REVIEW

File: riposte-metrics-codahale-signalfx/src/main/java/com/nike/riposte/metrics/codahale/contrib/SignalFxReporterFactory.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+     *                        .addUniqueDimension("host_foo", "host_bar")
+     *                        .setDetailsToAdd(SignalFxReporter.MetricDetails.ALL)
+     *      );
+     * </pre>
+     *
+     * <p>If you pass in null for the reporterConfigurator argument then {@link Function#identity()} will be used for
+     * a fully default SignalFx reporter.
+     *
+     * <p>You can adjust the {@link #getInterval()} and {@link #getTimeUnit()} methods if you want the SignalFx reporter
+     * to report on a different schedule than the 10 second default.
```
Comment (tlockney) [inline riposte-metrics-codahale-signalfx/src/main/java/com/nike/riposte/metrics/codahale/contrib/SignalFxReporterFactory.java]: Is it possible to adjust these without a code change? Should these be made configurable?

Comment (nicmunroe) [inline riposte-metrics-codahale-signalfx/src/main/java/com/nike/riposte/metrics/codahale/contrib/SignalFxReporterFactory.java]: I was following the pattern of the other ReporterFactory classes, and it's pretty trivial to create an anonymous extension class that overrides the method(s) if you want, but your comment made me think about it and I decided to make it configurable (at least for this one). Fixed.

**Comment by adriancole** [comment]
are y'all using signalfx now?

**Comment by nicmunroe** [comment]
@adriancole sorry for the delay responding to your question. Yes, SignalFx is one of the tools in our metrics gathering toolkit. :)

**Pull request merged**